src/01-dinosaur-facts.js

```diff
@@ -71,7 +94,29 @@ function getDinosaurDescription(dinosaurs, id) {}
  *  getDinosaursAliveMya(dinosaurs, 65, "unknown-key");
  *  //> ["WHQcpcOj0G"]
  */
-function getDinosaursAliveMya(dinosaurs, mya, key) {}
+function getDinosaursAliveMya(dinosaurs, mya, key) {
+  let results = [];
+  for (let i = 0; i < dinosaurs.length; i++) {
+     if (dinosaurs[i].mya[0]===mya){
+      results.push(dinosaurs[i].dinosaurId)
+    }if (dinosaurs[i].mya[1]===mya){
+      results.push(dinosaurs[i].dinosaurId)
+    }if (dinosaurs[i].mya[0]===mya-1){
+      results.push(dinosaurs[i].dinosaurId)
+    } if (dinosaurs[i].mya[1]===mya-1){
+      results.push(dinosaurs[i].dinosaurId)
+    } if (dinosaurs[i].mya[0]===mya && key){
+    results.push(dinosaurs[i].key)
+    }if (dinosaurs[i].mya[1]===mya && key){
+    results.push(dinosaurs[i].key)
+  }if (dinosaurs[i].mya[0]===mya-1 && key){
+    results.push(dinosaurs[i].key)
+  } if (dinosaurs[i].mya[1]===mya-1 && key){
+    results.push(dinosaurs[i].key)
+  }
+}return results
+}
+
```

Comment:
It looks like there's some misunderstanding here with the directions. 

You want your conditionals to check if the mya lies on or between the mya in each dinosaur to know if it was alive. 
this might look like:
```
if (dinosaurs[i].mya.length > 1) {
      if (dinosaurs[i].mya[0] >= mya && mya >= dinosaurs[i].mya[1]) { // if mya is between the first element and second element of the array > execute block
           if (key) {
             arr.push(dinosaurs[i][key])
           } else {
             arr.push(dinosaurs[i].dinosaurId)
           }
      }
}
```